File: Client/src/types/Equipment.ts
import { Booking } from './Booking';
import { Category } from './Category';
import { Location } from './Location';
import { EquipmentImage } from './EquipmentImage';

export type EquipmentCondition = 'Good' | 'Fair' | 'Poor' | 'Needs Repair';

export interface MaintenanceRecord {
    id: number;
    equipmentId: number;
    title: string;
    description: string;
    maintenanceDate: Date;
    performedBy: string;
    notes?: string;
    createdAt: Date;
    updatedAt?: Date;
}

export interface EquipmentResponseDto extends CreateEquipmentDto {
    id: string;
    ownerId: string;
    status: string;
    createdAt: Date;
    updatedAt?: Date;
    location: Location;
}

export interface Equipment {
    id: string;
    name: string;
    description: string;
    ownerId: string;
    category: Category;
    categoryId: number;
    tags: string[];
    condition: string;
    IsAvailable: boolean;
    locationId: string;
    location: Location;
    createdAt: string;
    updatedAt?: string;
    bookings: Booking[];
    maintenanceHistory: MaintenanceRecord[];
    images: EquipmentImage[];
}

// export interface Category {
//     id: number;
//     name: string;
//     iconName: string;
//     parentCategoryId?: number;
//     createdAt: Date;
//     updatedAt: Date;
// }

export interface CreateEquipmentDto {
    name: string;
    description: string;
    category: Category;
    condition: string;
    locationId: string;
    images?: EquipmentImage[];
}

export interface UpdateEquipmentDto extends CreateEquipmentDto {
    IsAvailable?: boolean;
} 

File: Server/Models/Equipment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Server.Models
{
    public class Equipment
    {
        [Key]
        public string Id { get; set; }

        [Required]
        [StringLength(100)]
        public string Name { get; set; }

        [Required]
        [StringLength(500)]
        public string Description { get; set; }

        [Required]
        public string OwnerId { get; set; }
        public ApplicationUser Owner { get; set; }

        public Category Category { get; set; }
        [Required]
        public int CategoryId { get; set; }

        public List<string> Tags { get; set; } = new List<string>();

        [Required]
        public string Condition { get; set; } = "Good";

        public bool IsAvailable { get; set; } = true;

        [Required]
        public string LocationId { get; set; }
        public Location Location { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        public List<Booking> Bookings { get; set; } = new List<Booking>();
        public List<MaintenanceRecord> MaintenanceHistory { get; set; } = new List<MaintenanceRecord>();
        public List<EquipmentImage> Images { get; set; } = new List<EquipmentImage>();
    }
}